### PR TITLE
Update TextEditingValue.selection on selection change

### PR DESCRIPTION
**Pull request by cbracken**

### CODE REVIEW

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 1]
```diff
@@ -381,6 +381,8 @@ class EditableTextState extends State<EditableText> implements TextInputClient {
   }
 
   void _handleSelectionChanged(TextSelection selection, RenderEditable renderObject, bool longPress) {
+    _value = _value.copyWith(selection: selection);
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/editable_text.dart]: How is this different from what happens on line 391?

Comment (abarth) [inline packages/flutter/lib/src/widgets/editable_text.dart]: Maybe we need to move line 391 here so its before `requestKeyboard`

**Comment by cbracken** [comment]
Fixes #9180

File: packages/flutter/lib/src/widgets/editable_text.dart (hunk 1 above)
Comment (cbracken) [inline packages/flutter/lib/src/widgets/editable_text.dart]: Done.

**Review by xster — APPROVED** [review]
Wow, the test framework is really good :)

File: packages/flutter/test/material/text_field_test.dart [hunk 2]
```diff
@@ -191,6 +191,39 @@ void main() {
...
+      ));
+    }
+
+    await tester.pumpWidget(builder());
+    expect(controller.selection.baseOffset, -1);
+    expect(controller.selection.extentOffset, -1);
+
+    final String testValue = 'abc def ghi';
+    await tester.enterText(find.byType(EditableText), testValue);
+    await tester.idle();
```
Comment (abarth) [inline packages/flutter/test/material/text_field_test.dart]: This idle shouldn't be needed.  It used to be needed, but at HEAD it is folded into enterText

Comment (cbracken) [inline packages/flutter/test/material/text_field_test.dart]: Sent #9290 that cleans up ever case I found in the framework tests.

Comment (abarth) [inline packages/flutter/test/material/text_field_test.dart]: Thanks!  I guess I missed a bunch of them before.

**Pull request merged**